internal/scheduler/scheduling/preempting_queue_scheduler_test.go

```diff
@@ -2104,7 +2113,8 @@ func TestPreemptingQueueScheduler(t *testing.T) {
 					)
 				}
 				for queue, qctx := range sctx.QueueSchedulingContexts {
-					assert.True(t, qctx.AllocatedByPriorityClass.Equal(allocatedByQueueAndPriorityClass[queue]))
+					m := internaltypes.RlMapFromJobSchedulerObjects(allocatedByQueueAndPriorityClass[queue], testfixtures.TestResourceListFactory)
+					assert.Equal(t, internaltypes.RlMapRemoveZeros(m), internaltypes.RlMapRemoveZeros(qctx.AllocatedByPriorityClass))
```

Comment:
The old `schedulerobjects` `Equal` methods treat missing and zero as the same. For now, reproduce the old behavior.